inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,175 @@ pheatmap(
+What do we interpret from the above regression analysis?
+
+1.The boxplot generated shows the distribution of Shannon diversity across
+different sample types. If the boxes are well-separated and there are significant 
+differences in medians, it indicates that SampleType has a strong effect on 
+diversity. This would be supported by significant p-values in the regression summary.
+2. The p-value associated with each coefficient tells us whether the effect
+of that sample type on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+3. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the sample type. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
+
+```{r}
+#| label: sub-checking-modeling-assumptions
+
+# Check Linear Modeling Assumptions
+# - Residual diagnostics to visually check for normality and homoscedasticity.
+# - Durbin-Watson Test for independence of residuals
+
+par(mfrow = c(2, 2))
+plot(model)
+dw_result <- dwtest(model)
+print(dw_result)
+```
+Here, 
+1. Residual diagnostics (normality and homoscedasticity):
+a. The Residuals vs Fitted plot provides insight into the relationship between residuals
+and fitted values. Ideally, the residuals should be randomly scattered around the horizontal
+line (residuals = 0) without any discernible pattern. This indicates that the model’s
+assumption of linearity and homoscedasticity (constant variance of residuals) is likely 
+met. If you observe any systematic patterns (such as a funnel shape), it may suggest
+heteroscedasticity, meaning the variance of residuals is not constant.
+b. The Normal Q-Q plot assesses whether the residuals are normally distributed. In this plot,
+residuals should align closely with the diagonal line. Deviations from this line, especially at
+the tails, indicate that the residuals are not normally distributed, which could violate the
+assumption of normality.
+c. The Scale-Location (or Spread-Location) plot also helps check the homoscedasticity assumption.
+It plots the square root of standardized residuals against the fitted values.
+d. The Residuals vs Leverage plot helps identify influential data points (outliers) that 
+might disproportionately affect the model. Points outside the dashed lines (Cook’s distance)
+are potential outliers that may have a strong influence on the model's coefficients.
+
+2. Durbin-Watson test checks for autocorrelation in the residuals. A value
+close to 2 indicates no autocorrelation. Values significantly different 
+from 2 suggest autocorrelation.
+
+In a nutshell, if all the assumptions hold (as indicated by the diagnostic plots
+and the statistical tests), you can be confident that the linear regression model
+is appropriate for the data, and the results can be interpreted reliably.
+
+Further, let's explore a continuous covariate, "coverage".
```

Comment:
What does this variable represent?

Reply:
I also thought to share more about the variable, shall add that in my upcoming commit.